pkg/flavor/flavor.go

```diff
@@ -80,6 +92,90 @@ func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.Virtu
 	return conflicts
 }
 
+func (m *methods) CheckFlavorRevision(vm *virtv1.VirtualMachine) error {
+	oldVM, err := m.client.VirtualMachine(vm.Namespace).Get(vm.Name, &v1.GetOptions{})
```

Comment:
while I don't think this validation is necessary (as I point out in a separate comment) you don't need to make an API call to get the previous VM. On an update to a previously created VM, the old VM and new VM are both in the request that's sent to the validation webhook.